### PR TITLE
chore: add entry verification error tests and fix migration checksums

### PR DESCRIPTION
**Pull request by pannpers**
## 🔗 Related Issue

Closes liverty-music/specification#14

## 📝 Summary of Changes

Follow-up to PR #79 (ZKP entry verification). Adds comprehensive error-path test coverage and fixes migration checksum issues.

### Error Propagation Tests
- Added 12 new test cases covering all error paths in entry use case:
  - `VerifyEntry`: merkle root retrieval, nullifier check, verifier error, insert error, oversized public signal
  - `GetMerklePath`: leaf index, root retrieval, path retrieval, leaf retrieval errors
  - `BuildMerkleTree`: list-by-event error, empty tickets, tree capacity overflow

### Migration Fixes
- Removed `-- +goose Down` sections from all 4 ticket system migration files — Atlas executes files as plain SQL and does not understand goose annotations, causing DOWN statements to run during UP migrations
- Regenerated `atlas.sum` checksum file

## 📋 Commit Log

```
301bf1c fix(db): remove goose DOWN sections incompatible with Atlas migration runner
341feb0 fix(db): regenerate atlas migration checksum after adding DOWN sections
8b0b84c test(entry): add error propagation and edge case tests
f920a6a chore(db): add goose DOWN rollback sections to ticket system migrations
```

## ✅ Self-Checklist

- [x] I have linked the related issue.
- [x] I have added or updated tests to cover my changes.
- [ ] I have updated the relevant documentation.
- [x] I have run `go test -race ./...` and `mise run lint` locally and all checks have passed.
- [x] My code follows the architecture and style guidelines of the project.

### CODE REVIEW

File: internal/infrastructure/database/rdb/migrations/versions/atlas.sum [hunk 1]
```diff
@@ -15,12 +15,12 @@ h1:r+nPUz0fMoilKhAFbWRySX/Thaz7Vfq2zZ6mlfqPuBU=
...
-20260221100000_add_ticket_system_columns.sql h1:asK8uLWGVpdGVLJZiCTh0fM8YpPnh8h/0IAm6C4jsuE=
-20260221100100_create_tickets_table.sql h1:DyBbRMHn6MRe+ynN+xqXT+DmeYGdZ33YHx13VrqScic=
-20260221100200_create_merkle_and_nullifiers.sql h1:YE+dxu9RK4mdpqwcizIO7QYeRLjQC4ogF2oPDBN1BRY=
-20260221120000_add_unique_index_on_artists_mbid.sql h1:EBNnKrQKAT5GRJN/mYLuQnZbhJSRbHe9p8thQdP7mss=
-20260221130000_drop_artists_timestamps.sql h1:yFpjhsJWaC8RmLsI+vfoRR1qWUkvtCA68pvKIkU3ZWo=
-20260221140000_add_check_constraints_non_empty_names.sql h1:USJ+CCmvYcaqeI82dFGjBh9QuUFqK9Py8x7tfAKYmGI=
-20260221150000_add_not_null_to_venues_raw_name.sql h1:g239yDe67KiU7XvF+r6ZF6fBas5WSxXQC40R83TKYk0=
-20260221160000_fix_ticket_system_schema.sql h1:C9M5d4XaQUQbw2QZADBlJl7lFyqGeAh45W9YGL+WhHs=
-20260222120000_add_push_subscriptions_table.sql h1:uNtEFTKmRotNO9I3uVUqBdpsjjdzfMq1kNIjNtsFNGI=
+20260221100000_add_ticket_system_columns.sql h1:e+BXO2pyU5C+rtUbz7awQ8tGC06XA+TrqkP1gUiZdVE=
```
[the remark this answers is not in the document]
Comment (pannpers) [reply internal/infrastructure/database/rdb/migrations/versions/atlas.sum]: The atlas.sum was regenerated correctly via `atlas migrate hash` after removing goose DOWN sections. Hash changes for unmodified files are expected — Atlas computes an integrity chain where each hash depends on all previous migrations. CI atlas-ci now passes.

File: internal/infrastructure/database/rdb/migrations/versions/20260221100000_add_ticket_system_columns.sql
[the remark this answers is not in the document]
Comment (pannpers) [reply internal/infrastructure/database/rdb/migrations/versions/20260221100000_add_ticket_system_columns.sql]: Good catch — updating the PR description now.
